### PR TITLE
LPS-167508 Move entity model

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket [LPS-167508](https://issues.liferay.com/browse/LPS-167508)
____
This is the first PR to work on the ticket link above.

We notice that the ObjectEntryEntityModel is on the module `object-rest-api` instead of in the module `object-rest-impl`
We decide to move it back to the `*-impl` module because:
* It makes more sense to be in the `*-impl` module as is the implementation of the `EntityModel` interface
* All the EntityModels are placed in their corresponding `*-impl` modules, so, for consistency it should be there.
* To evolve the functionality, it is easier to be in a `*-impl` module.

The PR makes that move and adapts the test related to the ObjectEntryEntityModel.

The most important changes are:
* The change to inject the `objectDefinition` in `ObjectEntryResourceImpl` through the constructor instead of the `Context`. As we were founding some issues to inject this `objectDefinition`, we think that this way is better to avoid those problems as now we use the `PrototypeServiceFactory` in the `Deployer`.
* The adaptation of the test to get the entityModel from the ObjectEntryResourceImpl instance instead of instance it directly. In this way, it is possible to test the entityModel generated by the Resource and check if the entityFields are the expected or not

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
It seems the test is failing

**Comment by LuismiBarcos** [comment]
Weird, locally is not failing and I trying to reproduce it with the same environment as the CI

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
Fixed conflicts and avoid the idea of injecting the `_objectDefinition` through the constructor.
Instead, I changed the test to inject the `_objectDefinition` with reflexion to avoid changing the business code in ObjectEntryResourceImpl.

cc/ @ccorreagg FYI if you want to review it. I will send it tomorrow if the CI is ok

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/odata/entity/v1_0/test/ObjectEntryEntityModelTest.java [hunk 1]
```diff
@@ -154,9 +158,14 @@ public void testGetEntityFieldsMap() throws PortalException {
...
 			).putAll(
-				_getEntityFieldsMap(customObjectFields)
+				_getExpectedEntityFieldsMap(
+					customObjectFields, relatedObjectDefinition,
+					objectRelationship)
 			).build(),
-			objectEntryEntityModel.getEntityFieldsMap());
+			_getObjectDefinitionEntityFieldsMap(objectDefinition));
+
+		_deleteCreatedData(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/odata/entity/v1_0/test/ObjectEntryEntityModelTest.java:167]: NOTE: If you includes this code in here and the test throws a exception, the created data will stays in the database

Comment (LuismiBarcos) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/odata/entity/v1_0/test/ObjectEntryEntityModelTest.java]: Good catch!!

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
The test is failing on the CI 😩

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
Test failures unrelated. OK to manually forward ✅

**Comment by LuismiBarcos** [comment]
Sent manually -> https://github.com/brianchandotcom/liferay-portal/pull/126739

**Pull request closed without merging**